### PR TITLE
Support ZINTERCARD

### PR DESCRIPTION
**Pull request by Avital-Fine**
Documentation of the command: https://redis.io/commands/zintercard

Also, the use of the names `keys` and `sets` was confusing. 
I think it makes sense to change all the keys to sets.
Let me know what do you think?

### CODE REVIEW

**Review by sazzad16 — CHANGES_REQUESTED** [review]
Undo renaming all `keys` to `sets`, conforming redis.io doc.

File: src/main/java/redis/clients/jedis/MultiNodePipelineBase.java [hunk 1]
```diff
@@ -1118,6 +1118,16 @@ public final void sync() {
     return appendCommand(commandObjects.zinterWithScores(params, keys));
   }
 
+  @Override
+  public Response<Long> zintercard(String... sets) {
+    return appendCommand(commandObjects.zintercard(keys));
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/MultiNodePipelineBase.java:1123]:
```
[ERROR] COMPILATION ERROR : 
[ERROR] /home/circleci/project/src/main/java/redis/clients/jedis/MultiNodePipelineBase.java:[1123,52] cannot find symbol
  symbol:   variable keys
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]
ZINTERCARD numkeys key [key ...] [LIMIT limit] 

@Avital-Fine As long as redis.io using key/keys, we're also using key/keys.

File: src/main/java/redis/clients/jedis/CommandObjects.java [hunk 2]
```diff
@@ -1678,6 +1678,16 @@ protected CommandArguments commandArguments(ProtocolCommand command) {
         .addParams(params).add(WITHSCORES), BuilderFactory.TUPLE_ZSET);
   }
 
+  public final CommandObject<Long> zintercard(String... sets) {
```
Comment (zeekling) [inline src/main/java/redis/clients/jedis/CommandObjects.java:1681]:
```suggestion
  public final CommandObject<Long> zintercard(String... keys) {
```

File: src/main/java/redis/clients/jedis/commands/SortedSetBinaryCommands.java [hunk 3]
```diff
@@ -142,6 +142,10 @@
 
   long zinterstore(byte[] dstkey, ZParams params, byte[]... sets);
 
+  long zintercard(byte[]... sets);
+
+  long zintercard(long limit, byte[]... sets);
```
Comment (zeekling) [inline src/main/java/redis/clients/jedis/commands/SortedSetBinaryCommands.java:147]: Please add javadoc for interface.

File: src/main/java/redis/clients/jedis/commands/SortedSetCommands.java [hunk 4]
```diff
@@ -143,6 +143,10 @@
 
   Set<Tuple> zinterWithScores(ZParams params, String... keys);
 
+  long zintercard(String... sets);
+
+  long zintercard(long limit, String... sets);
```
Comment (zeekling) [inline src/main/java/redis/clients/jedis/commands/SortedSetCommands.java:148]: Please add javadoc for interface.

File: src/main/java/redis/clients/jedis/commands/SortedSetPipelineBinaryCommands.java [hunk 5]
```diff
@@ -143,6 +143,10 @@
 
   Response<Long> zinterstore(byte[] dstkey, ZParams params, byte[]... sets);
 
+  Response<Long> zintercard(byte[]... sets);
+
+  Response<Long> zintercard(long limit, byte[]... sets);
```
Comment (zeekling) [inline src/main/java/redis/clients/jedis/commands/SortedSetPipelineBinaryCommands.java:148]: Please add javadoc for interface.

File: src/main/java/redis/clients/jedis/commands/SortedSetPipelineCommands.java [hunk 6]
```diff
@@ -144,6 +144,10 @@
 
   Response<Set<Tuple>> zinterWithScores(ZParams params, String... keys);
 
+  Response<Long> zintercard(String... sets);
+
+  Response<Long> zintercard(long limit, String... sets);
```
Comment (zeekling) [inline src/main/java/redis/clients/jedis/commands/SortedSetPipelineCommands.java:149]: Please add javadoc for interface.

**Review by zeekling — CHANGES_REQUESTED** [review]

File: src/main/java/redis/clients/jedis/CommandObjects.java [hunk 7]
```diff
@@ -1678,6 +1678,16 @@ protected CommandArguments commandArguments(ProtocolCommand command) {
         .addParams(params).add(WITHSCORES), BuilderFactory.TUPLE_ZSET);
   }
 
+  public final CommandObject<Long> zintercard(String... keys) {
+    return new CommandObject<>(commandArguments(ZINTERCARD)
+            .add(keys.length).keys((Object[]) keys), BuilderFactory.LONG);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/CommandObjects.java:1683]:
```suggestion
    return new CommandObject<>(commandArguments(ZINTERCARD).add(keys.length)
        .keys((Object[]) keys), BuilderFactory.LONG);
```

File: src/main/java/redis/clients/jedis/CommandObjects.java [hunk 8]
```diff
@@ -1678,6 +1678,16 @@ protected CommandArguments commandArguments(ProtocolCommand command) {
...
+  }
+
+  public final CommandObject<Long> zintercard(long limit, String... keys) {
+    return new CommandObject<>(commandArguments(ZINTERCARD)
+            .add(keys.length).keys((Object[]) keys).add(LIMIT).add(limit), BuilderFactory.LONG);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/CommandObjects.java:1688]:
```suggestion
    return new CommandObject<>(commandArguments(ZINTERCARD).add(keys.length)
        .keys((Object[]) keys).add(LIMIT).add(limit), BuilderFactory.LONG);
```

File: src/main/java/redis/clients/jedis/CommandObjects.java [hunk 9]
```diff
@@ -1688,6 +1698,16 @@ protected CommandArguments commandArguments(ProtocolCommand command) {
         .add(sets.length).keys((Object[]) sets).addParams(params), BuilderFactory.LONG);
   }
 
+  public final CommandObject<Long> zintercard(byte[]... keys) {
+    return new CommandObject<>(commandArguments(ZINTERCARD)
+        .add(keys.length).keys((Object[]) keys), BuilderFactory.LONG);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/CommandObjects.java:1703]:
```suggestion
    return new CommandObject<>(commandArguments(ZINTERCARD).add(keys.length)
        .keys((Object[]) keys), BuilderFactory.LONG);
```

File: src/main/java/redis/clients/jedis/CommandObjects.java [hunk 10]
```diff
@@ -1688,6 +1698,16 @@ protected CommandArguments commandArguments(ProtocolCommand command) {
...
+  }
+
+  public final CommandObject<Long> zintercard(long limit, byte[]... keys) {
+    return new CommandObject<>(commandArguments(ZINTERCARD)
+            .add(keys.length).keys((Object[]) keys).add(LIMIT).add(limit), BuilderFactory.LONG);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/CommandObjects.java:1708]:
```suggestion
    return new CommandObject<>(commandArguments(ZINTERCARD).add(keys.length)
        .keys((Object[]) keys).add(LIMIT).add(limit), BuilderFactory.LONG);
```

File: src/main/java/redis/clients/jedis/Protocol.java [hunk 11]
```diff
@@ -235,7 +235,7 @@ public static Object read(final RedisInputStream is) {
     GEORADIUSBYMEMBER, GEORADIUSBYMEMBER_RO, BITFIELD, HSTRLEN, TOUCH, SWAPDB, MEMORY, XADD, XLEN,
     XDEL, XTRIM, XRANGE, XREVRANGE, XREAD, XACK, XGROUP, XREADGROUP, XPENDING, XCLAIM, XAUTOCLAIM,
     XINFO, BITFIELD_RO, LPOS, ZMSCORE, BZPOPMIN, BZPOPMAX, BLMOVE, LMOVE, COPY, ROLE, FAILOVER,
-    STRALGO, GEOSEARCH, GEOSEARCHSTORE, LOLWUT, REPLICAOF, ZRANGESTORE, SINTERCARD, SORT_RO;
+    STRALGO, GEOSEARCH, GEOSEARCHSTORE, LOLWUT, REPLICAOF, ZRANGESTORE, SINTERCARD, SORT_RO, ZINTERCARD;
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/Protocol.java:238]:
```suggestion
    STRALGO, GEOSEARCH, GEOSEARCHSTORE, LOLWUT, REPLICAOF, ZRANGESTORE, SINTERCARD, ZINTERCARD,
    SORT_RO;
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]

**Review by sazzad16 — APPROVED** [review]

**Review by sazzad16 — APPROVED** [review]

**Review by dengliming — APPROVED** [review]

**Pull request merged**